### ISSUE TITLE
Generate pseudocode (sema_seg) with all black background images and Deeplab segmentation that only recognizes the background

### ISSUE THREAD

**Issue by zbb1111** [zhaozhengChen/ReCAM#26]
Hello, thank you very much for your excellent article. I have some confusion and I hope you can help me answer it:
1. The pseudocode (sema_seg XX. png) generated by Recam (10582 images) is all black background images, but the miou in the log is 0.7. Then, I used these 10582 images to replace the SegmentationAugClass (12031) in Deeplab for training, but the final test result was only 0.03, only recognizing the background.
I am a beginner, and there may be some areas where I misunderstood. I hope you can answer them for me. Thank you!
![微信图片_20240414225053](https://github.com/zhaozhengChen/ReCAM/assets/128360072/9f22d623-ca84-445b-b518-80b887abdf53)
![2](https://github.com/zhaozhengChen/ReCAM/assets/128360072/3fad5f12-f522-4f11-b448-affe260b0956)
![3](https://github.com/zhaozhengChen/ReCAM/assets/128360072/f7b25dc8-7d45-42a0-a059-373cd238e776)

**Comment by zhaozhengChen** [comment]
Thanks for your interest in our work.
1. As the mIoU is correct, I think the reason is that the label range in the png file is 0-20 (VOC), which is close to 0. Then the color should be close to black.
2. Is the training loss of deeplab normal (not NaN)? In my experiment, I observed that the loss may not be stable.

**Comment by zbb1111** [comment]
Yes,the training loss of deeplab normal is NaN,What is the reason for this? How should I solve it?

**Comment by zbb1111** [comment]
In addition，Can I take a look at the result graph of the generated visual pseudocode, sema_seg?

**Comment by zbb1111** [comment]
I'm sorry to trouble you with so many questions. I hope you can reply to me when you have time. Sincere thanks!!

I found that Deeplab stopped at 99 iterations during training. Is this normal? I have run it several times and it's like this. Can this be the reason for the loss NAN
![Uploading 微信图片_20240414232621.jpg…]()

**Comment by zbb1111** [comment]
I'm sorry to trouble you with so many questions. I hope you can reply to me when you have time. Sincere thanks!!

I found that Deeplab stopped at 99 iterations during training. Is this normal? I have run it several times and it's like this. Can this be the reason for the loss NAN
![微信图片_20240414232621](https://github.com/zhaozhengChen/ReCAM/assets/128360072/bd0b92eb-2923-4536-a5ac-1e6fe7b9d6f1)

**Comment by zhaozhengChen** [comment]
1. For the Nan problem, in most cases, rerun works. In the Figure you provided, it does not stop at 99 iterations. It seems you got a warning, I think it's not the reason.
2. For visualization, you can refer to some visualization code like [this](https://github.com/cao-cong/ADS-SemiSeg/blob/5fe9d5734125d97a26914722500a34472d5688a5/evaluate.py).

**Comment by zhaozhengChen** [comment]
The Nan problem always occurs in the first few iterations. You can observe the loss of the first few iterations to judge whether it is running normally.

**Comment by zbb1111** [comment]
I have solved this problem, thank you very much!!!
Wishing you all the best!
